src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -0,0 +1,141 @@
+import styled from 'styled-components'
+
+import { Color, theme } from '../theme'
+import { Box } from '../Box'
+import { Tag } from '../Tag'
+import { Text } from '../Text'
+import React from 'react'
+import { ToggleButton } from './ToggleButton'
+
+export type SegmentedControlOption<T> = {
+  label: string
+  value: T
+  /** optional tag able to show in the toggle if showTag prop is passed */
+  tag?: string
+  isDisabled?: boolean
+}
+
+export type SegmentedControlProps<T> = {
+  /** you can pass more than 2 options */
+  options: SegmentedControlOption<T>[]
+  value: T
+  onChange: (value: T) => void
+  /**  pass this prop if you want to show the tag in the toggle */
+  showTag?: boolean
+  toggleColor?: Color
+  backgroundColor?: Color
+  textColor?: Color
+  selectedTextColor?: Color
+  tagColor?: Color
+  tagTextColor?: Color
+  tagBorderColor?: Color
+}
+
+/**recommend to use a useState to store the selected value above the component */
+export const SegmentedControl = <T,>({
+  options,
+  value,
+  onChange,
+  showTag,
+  toggleColor: toggleColor,
+  backgroundColor,
+  textColor = 'liquorice',
+  selectedTextColor = 'cream',
+  tagColor = 'marshmallowPink',
+  tagBorderColor = 'marshmallowPink',
+  tagTextColor = 'cream',
+}: SegmentedControlProps<T>) => {
+  return (
+    <ToggleWrapper backgroundColor={backgroundColor}>
+      <IndicatorWrapper backgroundColor={backgroundColor}>
+        {options.map((option) => {
+          return (
+            <ToggleButton
+              key={option.value as string}
```

Comment:
I don't think its a good point to get rid of the generic types, because the map key don't accept generic type. a better solution here is to change the key from value to label.

Reply:
Using label solves this issue yeah